internal/setup/setup.go

```diff
@@ -375,27 +395,46 @@ func (yp *yamlPath) SameContent(other *yamlPath) bool {
 }
 
 type yamlArch struct {
-	list []string
+	List []string
 }
 
 func (ya *yamlArch) UnmarshalYAML(value *yaml.Node) error {
 	var s string
 	var l []string
 	if value.Decode(&s) == nil {
-		ya.list = []string{s}
+		ya.List = []string{s}
 	} else if value.Decode(&l) == nil {
-		ya.list = l
+		ya.List = l
 	} else {
 		return fmt.Errorf("cannot decode arch")
 	}
 	// Validate arch correctness later for a better error message.
 	return nil
 }
 
+func (ya yamlArch) MarshalYAML() (interface{}, error) {
+	if len(ya.List) == 1 {
+		return ya.List[0], nil
+	}
+	return ya.List, nil
+}
+
+type yamlMode uint
+
+func (ym yamlMode) MarshalYAML() (interface{}, error) {
+	// Workaround for marshalling integers in octal format.
+	// Ref: https://github.com/go-yaml/yaml/issues/420.
+	return &yaml.Node{
+		Kind:  yaml.ScalarNode,
+		Tag:   "!!int",
+		Value: fmt.Sprintf("0%o", ym),
+	}, nil
```

Comment:
I think this will be more robust for the future if we marshall the integer into a node and then change the value. That way we do not have to fill all the values manually, but I don't have a strong opinion. Thoughts?

Reply:
That does sound better actually.

Reply:
Changed it, please take another look!